airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlJdbcPartitionFactory.kt

```diff
@@ -51,13 +77,26 @@ class MysqlJdbcPartitionFactory(
                     streamState,
                 )
             }
-            return MysqlJdbcSnapshotPartition(
-                selectQueryGenerator,
-                streamState,
-                pkChosenFromCatalog,
-                lowerBound = null,
-                upperBound = null,
-            )
+
+            val upperBound = findPkUppderBound(stream, pkChosenFromCatalog)
+
+            if (sharedState.configuration.global) {
+                return MysqlJdbcCdcRfrSnapshotPartition(
+                    selectQueryGenerator,
+                    streamState,
+                    pkChosenFromCatalog,
+                    lowerBound = null,
+                    upperBound = listOf(upperBound)
+                )
+            } else {
+                return MysqlJdbcRfrSnapshotPartition(
```

Comment:
Is this correct for non global incremental ? i.e user cursor incremental partition?